src/Workspaces/Core/Portable/Utilities/BKTree.Edge.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Roslyn.Utilities
+{
+    internal partial class BKTree
+    {
+        private struct Edge
+        {
+            // The edit distance between the child and parent connected by this edge.
```

Comment:
Consider: explain that the parent node is based on the index in the containing array at which this Edge appears.